### PR TITLE
Fixing the "Publish to Demo" Workflow

### PR DESCRIPTION
**Pull request by mirrodi**
### Description
The workflow run every 3 weeks. Therefore a modulo function is needed to determine the weeks passed.

### Reference

Issues: - (Bug Fix)

### Definition of Done
- [x] Acceptance criteria are met
- [x] Build/Test workflow has successfully finished
- [ ] Release notes are complemented
- [ ] Documentation is complemented (operator manual, system specification)
- [ ] Frontend is locally smoke-tested
- [x] Side branches are deleted
- [ ] Board is updated
- [x] Infrastructure is adjusted

### CODE REVIEW

File: .github/workflows/publish-demo.yml [hunk 1]
```diff
@@ -22,11 +22,11 @@ jobs:
       - name: Get number of weeks passed
         id: get-number-of-weeks
         run: | # 604800 is 1 week in seconds
-          number_of_weeks=$((${{ steps.calculate-time-passed.outputs.time_passed }}/604800))
+          number_of_weeks=$(((${{ steps.calculate-time-passed.outputs.time_passed }}%3)/604800)) # replace the 3 with the length of one sprint or the desired period duration
```
Comment (martind260) [inline .github/workflows/publish-demo.yml:25]: Bevor du einen so langen Kommentar schreibst, kannst Du auch gleich eine Variable SPRING_LENGTH_IN_WEEKS=3 anlegen.

Comment (martind260) [inline .github/workflows/publish-demo.yml:25]: NumberOfWeeks ist doch time_passed / 604800

File: .github/workflows/publish-demo.yml [hunk 2]
```diff
@@ -22,11 +22,11 @@ jobs:
...
           echo "::set-output name=number_of_weeks::$number_of_weeks"
 
       - name: Check if job should be run # the automatically triggered actions should only be executed every third week
-        if: github.event_name != 'workflow_dispatch' && steps.get-number-of-weeks.outputs.number_of_weeks != 3 # change!! the period duration by changing the number of weeks to pass
+        if: github.event_name != 'workflow_dispatch' && steps.get-number-of-weeks.outputs.number_of_weeks != 0
```
Comment (martind260) [inline .github/workflows/publish-demo.yml]: Zur Berechnung brauchst Du doch: (time_passed / weekInSecs) % SPRINT_LENGHT_IN_WEEKS

Comment (mirrodi) [reply .github/workflows/publish-demo.yml]: ah, ja stimmt. Hab die Formel verwechselt

**Review by martind260 — APPROVED** [review]

**Pull request merged**